### PR TITLE
Add chezmoi

### PR DESCRIPTION
**Pull request by Randomblock1**
Add chezmoi from https://github.com/twpayne/chezmoi
Needs Go 1.16 (#579), and it compiles just fine on my iPhone and ARM Mac.

### CODE REVIEW

**Comment by Diatrus** [comment]
Going to wait to merge until I do golang 1.16. Should hopefully be merged today. 👍

File: chezmoi.mk [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	@echo "Using previously built chezmoi."
+else
+chezmoi: chezmoi-setup
+	cd $(BUILD_WORK)/chezmoi && go get -d -v .
+	cd $(BUILD_WORK)/chezmoi && \
+		CGO_ENABLED=1 \
+		CGO_CFLAGS="$(CFLAGS)" \
+		CGO_CPPFLAGS="$(CPPFLAGS)" \
+		CGO_LDFLAGS="$(LDFLAGS)" \
+		CC=cc \
```
Comment (quiprr) [inline chezmoi.mk:37]: Shouldn't this be $(CC)?

**Review by quiprr — CHANGES_REQUESTED** [review]

File: chezmoi.mk (hunk 1 above)
Comment (Randomblock1) [reply chezmoi.mk:37]: Whoops, yes it should be.

**Comment by Diatrus** [comment]
@Randomblock1 would you mind updating this PR to use the new DEFAULT_GOLANG_FLAGS variable from commit f63bcf3216d8a31f8454d17c6018657677220aac? It should remove the need for any of that GO_ARGS stuff.

**Comment by Randomblock1** [comment]
OK, it should work with the new variables now.

**Comment by Diatrus** [comment]
Thank you

**Pull request merged**